datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -1420,6 +1421,73 @@ fn contain(e: &Expr, check_map: &HashMap<String, Expr>) -> bool {
     is_contain
 }
 
+/// Infers new predicates by substituting equalities.
+/// For example, with predicates `t2.b = 3` and `t1.b > t2.b`,
+/// we can infer `t1.b > 3`.
+fn infer_predicates_from_equalities(predicates: Vec<Expr>) -> Result<Vec<Expr>> {
```

Comment:
Related to:
https://github.com/apache/datafusion/pull/15906

Should we reopen our upstream PR?

Reply:
Will do later